File: solutionFiles/TwoSum.py
import sys
import json

def two_sum(nums, target):
    num_dict = {}

    for i, num in enumerate(nums):
        complement = target - num
        if complement in num_dict:
            result = [num_dict[complement], i]
            return json.dumps(result)
        num_dict[num] = i

    return json.dumps([])  # In case no solution is found (though the problem guarantees a solution)

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python script.py '{\"nums\": [...], \"target\": ...}'")
        sys.exit(1)

    input_json = sys.argv[1]
    try:
        input_data = json.loads(input_json)
        nums = input_data["nums"]
        target = input_data["target"]
        print(two_sum(nums, target))
    except (json.JSONDecodeError, KeyError):
        print("Invalid input format. Please provide a JSON string with 'nums' and 'target'.")

File: solutionFiles/TwoSum.js
const fs = require('fs');

function twoSum(nums, target) {
    const numDict = {};

    for (let i = 0; i < nums.length; i++) {
        const num = nums[i];
        const complement = target - num;
        if (complement in numDict) {
            return JSON.stringify([numDict[complement], i]);
        }
        numDict[num] = i;
    }

    return JSON.stringify([]);  // In case no solution is found (though the problem guarantees a solution)
}

if (process.argv.length !== 3) {
    console.log("Usage: node script.js '{\"nums\": [...], \"target\": ...}'");
    process.exit(1);
}

const inputJson = process.argv[2];
try {
    const inputData = JSON.parse(inputJson);
    const nums = inputData.nums;
    const target = inputData.target;
    console.log(twoSum(nums, target));
} catch (e) {
    console.log("Invalid input format. Please provide a JSON string with 'nums' and 'target'.");
}
